src/coreclr/vm/jitinterface.cpp

```diff
@@ -5945,6 +5987,40 @@ CorInfoHelpFunc CEEInfo::getUnBoxHelper(CORINFO_CLASS_HANDLE clsHnd)
     return CORINFO_HELP_UNBOX;
 }
 
+/***********************************************************************/
+void* CEEInfo::getRuntimeTypePointer(CORINFO_CLASS_HANDLE clsHnd)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    void* pointer = nullptr;
+
+    JIT_TO_EE_TRANSITION();
+
+    GCX_COOP();
+
+    TypeHandle typeHnd(clsHnd);
+    if (!typeHnd.IsNull() && !typeHnd.IsTypeDesc())
```

Comment:
You have fixed the typedescs. Why not take advantage of it here?

Also, I would omit the null check. The JIT should not be passing null type handles to JIT/EE interface.

Reply:
Makes sense, although my app I tested this PR on (~100k methods to jit) hit this path with TypeDesc only once

Reply:
I'll have to apply this patch https://github.com/EgorBo/runtime-1/commit/908dc940df6201658d124624dd7373df2cd82fd5
but considering minimal impact on diff perhaps it's better to keep as is?

Reply:
You should not need to duplicate the logic for getting the managed class object. Can it look like this?

```
void* CEEInfo::getRuntimeTypePointer(CORINFO_CLASS_HANDLE clsHnd)
{
    CONTRACTL{
        THROWS;
        GC_TRIGGERS;
        MODE_PREEMPTIVE;
    } CONTRACTL_END;

    void* pointer = nullptr;

    JIT_TO_EE_TRANSITION();

    TypeHandle typeHnd(clsHnd);
    if (!typeHnd.IsCanonicalSubtype() && typeHnd.IsManagedClassObjectPinned())
    {
        GCX_COOP();
        pointer = OBJECTREFToObject(typeHnd.GetManagedClassObject());
    }
    
    EE_TO_JIT_TRANSITION();

    return pointer;
}

bool TypeHandle::IsManagedClassObjectPinned()
{
    LIMITED_METHOD_CONTRACT;

    // Function pointers are always mapped to typeof(IntPtr)
    return !GetLoaderAllocator()->CanUnload() || IsFnPtrType();
}

// The AllocateRuntimeTypeObject helper method can live here as well. MethodTable is probably not the best place for it.

void TypeHandle::AllocateManagedClassObject(RUNTIMETYPEHANDLE* pDest)
{
    CONTRACTL {
        THROWS;
        GC_TRIGGERS;
        MODE_COOPERATIVE;
    } CONTRACTL_END;

    LoaderAllocator* pLoaderAllocator = GetLoaderAllocator();
    if (!pLoaderAllocator->CanUnload())
    {
...
    }
    else
    {
...
    }
}
```